docker/docker_client.go

```diff
@@ -42,8 +47,13 @@ type dockerClient struct {
 	transport       *http.Transport
 }
 
-// newDockerClient returns a new dockerClient instance for refHostname (a host a specified in the Docker image reference, not canonicalized to dockerRegistry)
-func newDockerClient(refHostname, certPath string, tlsVerify bool) (*dockerClient, error) {
+// NewClient returns a new Client instance for refHostname (a host a specified in the Docker image reference, not canonicalized to dockerRegistry)
+func NewClient(img, certPath string, tlsVerify bool) (Client, error) {
+	ref, _, err := parseDockerImageName(img)
```

Comment:
This is not so clean here, I would move the `parseDockerImageName` call up in the call stack but that can affect `NewImageSource` as well, not sure. I believe we can live with this duplication but since `NewClient` is exported I'd rather have a better API here.